### PR TITLE
Use Static public addresses, allocate additional public addresses to simulate web apps.

### PR DESCRIPTION
**Pull request by chkp-zegman**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@chkp-zegman__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: checkpoint-multi-nic/azuredeploy.parameters.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
             "value": "changeme"
```
Comment (singhkay) [inline checkpoint-multi-nic/azuredeploy.parameters.json]: @chkp-zegman Can you look at putting in placeholder parameters that will allow us to test this template with our CI? https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#travis-ci

File: checkpoint-single-nic/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            },
+            "type": "string"
+        }
+    },
+    "variables": {
+        "apiVersion": "2015-06-15",
+        "customData": "#!/bin/bash",
+        "addressPrefix": "10.0.0.0/16",
+        "gatewayPrivateAddress": "10.0.1.10",
+        "onPremiseNetwork": "198.51.100.0/24",
```
Comment (singhkay) [inline checkpoint-single-nic/azuredeploy.json]: Is this the same for everyone?

File: checkpoint-multi-nic/azuredeploy.parameters.json (hunk 1 above)
Comment (chkp-zegman) [inline checkpoint-multi-nic/azuredeploy.parameters.json]: Should now be fixed.

File: checkpoint-single-nic/azuredeploy.json (hunk 2 above)
Comment (chkp-zegman) [inline checkpoint-single-nic/azuredeploy.json]: Not sure I understand your comment, could you elaborate?
To which piece of code are you referring?

Comment (singhkay) [inline checkpoint-single-nic/azuredeploy.json]: The "onPremiseNetwork": "198.51.100.0/24" ip

Comment (chkp-zegman) [inline checkpoint-single-nic/azuredeploy.json]: This is just a placeholder, and the customer could replace it with his own.
Note that 198.51.100.0/24 is part of an address range reserved for documentation.
See: https://tools.ietf.org/html/rfc5737

Comment (singhkay) [inline checkpoint-single-nic/azuredeploy.json]: Ah ok, got it. In that case it looks like this template needs to have some existing infrastructure setup before being able to run succesfully?

File: checkpoint-single-nic/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+                        "name": "External-Subnet",
+                        "properties": {
+                            "addressPrefix": "[variables('subnet1Prefix')]",
+                            "nextHopType": "VirtualAppliance",
+                            "nextHopIpAddress": "[variables('gatewayPrivateAddress')]"
+                        }
+                    }
+                ]
+            },
+            "type": "Microsoft.Network/routeTables"
```
Comment (singhkay) [inline checkpoint-single-nic/azuredeploy.json]: Can you put the resource types at the top? It's hard to identify the resources with the types all the way at the bottom

File: checkpoint-multi-nic/custom-data.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
```
Comment (singhkay) [inline checkpoint-multi-nic/custom-data.py]: What is the purpose of this python script?

File: checkpoint-multi-nic/azuredeploy.json [hunk 5]
```diff
@@ -27,10 +27,10 @@
         },
```
Comment (singhkay) [inline checkpoint-multi-nic/azuredeploy.json]: Cosmetic suggestion: It's easier to read and review with 2 space tabs :)

File: checkpoint-single-nic/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+            "offer": "[variables('imageOffer')]",
+            "publisher": "[variables('imagePublisher')]",
+            "sku": "sg-ngtp",
+            "version": "latest"
+        },
+        "nicName": "gateway-nic",
+        "osProfile": "[variables(concat('osprofile',parameters('authenticationType')))]",
+        "osProfilepassword": {
+            "adminPassword": "[parameters('adminPassword')]",
+            "adminUsername": "notused",
```
Comment (singhkay) [inline checkpoint-single-nic/azuredeploy.json]: Does the user use this username to authenticate to the VM?

File: checkpoint-single-nic/azuredeploy.json (hunk 3 above)
Comment (chkp-zegman) [inline checkpoint-single-nic/azuredeploy.json]: Done.

File: checkpoint-multi-nic/azuredeploy.json (hunk 5 above)
Comment (chkp-zegman) [inline checkpoint-multi-nic/azuredeploy.json]: Done.

**Comment by chkp-zegman** [comment]
Hi,

I noticed that CI fails. I suspect this might be related to one of two things:
- Our templates launch an image from the marketplace. Normally the user needs to enable programmatic access to the image by first agreeing to the terms and conditions in the Azure portal.
- Travis seems to generate a longer than allowed password (23 characters instead of the maximum which is 16).
In any case, it is not clear to me whether this is blocking our pull request or not.

File: checkpoint-single-nic/azuredeploy.json (hunk 6 above)
Comment (chkp-zegman) [inline checkpoint-single-nic/azuredeploy.json]: The admin user name is always admin. 
You need to pass a user name in the profile but our image always disregards it and sets a user called admin.

File: checkpoint-multi-nic/custom-data.py (hunk 4 above)
Comment (chkp-zegman) [inline checkpoint-multi-nic/custom-data.py]: It reads an initialization script and encodes it into the custom data in azure deploy.json.

**Comment by singhkay** [comment]
@chkp-zegman I enabled programattic access to the image from Portal for our CI subscription so that should be taken care of. I haven't heard about the 16 char password limit before. The Travis CI gives you the exact template it deployed, if you replace the 23 char password in it with a 16 char one, does the template deploy?

cc: @sedouard

**Comment by chkp-zegman** [comment]
@singhkay 
We managed to deploy the template (with the original password provided by Travis CI). 
We also managed to SSH into the VM using the password.
Please note that as part of this template the VM runs a configuration script after which it reboots automatically.
Questions:
- Is the CI failure blocking the pull request, if not I kindly request that you move forward with it.
- Is there a way to receive additional information on the cause of the CI failure?

File: checkpoint-multi-nic/azuredeploy.parameters.json [hunk 7]
```diff
@@ -1,27 +1,39 @@
...
-        "sshPublicKey": {
-            "value": ""
-        },
-        "vmName": {
-            "value": "changeme"
-        },
-        "vmSize": {
-            "value": "Standard_D2"
-        }
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
```
Comment (singhkay) [inline checkpoint-multi-nic/azuredeploy.parameters.json]: This should be "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#"

File: checkpoint-single-nic/azuredeploy.parameters.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
```
Comment (singhkay) [inline checkpoint-single-nic/azuredeploy.parameters.json]: This should be "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",

**Comment by singhkay** [comment]
@chkp-zegman Looks like I enabled programmatic access for the wrong checkpoint images. All good now. Left a comment for a minor change, can merge once updated.

**Comment by chkp-zegman** [comment]
Done.

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**